### ISSUE TITLE
Joomla Content Editor: Resizing the WYSIWYG text area in the Joomla editor can only be done by dragging

### ISSUE THREAD

**Issue by mbeganyi-a11y** [joomla/joomla-cms#48425]
### Steps to reproduce the issue

**NOTE**: Accessibility backlog issue and work-in-progress. This issue has been opened to account for all the issues flagged in the initial [Milestone 1a Audit Report](https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report). If this is a duplicate issue, it will be closed by the end of September 2026. 

Milestone 1a   finding ID | M1a-019
-- | --
Backlog   priority | Low
Severity —   carried from Milestone 1a | Minor
Implementation   effort | Low
Target   milestone(s) | MS2a, MS3a,   MS3b, MS7b, MS8a, MS10
WCAG 2.2   success criterion | 2.5.7   Dragging Movements
Responsibility   tag(s) from report | Design,   Development
Affected   Joomla area/component | https://accessibility-project.joomla.org/results/Milestone-1a/milestone1/deliverable1/wcag-report#fe22f934-2320-4b64-91f1-46fbd9e9a236
Affected user   journey(s) | 09 – Backend   Edit Form Pattern; 10 – Frontend User Forms
Source report/reference | Milestone 1a   WCAG audit report
STF reporting   reference | To be   assigned
Issue owner | To be   assigned
Status | Triage

<p class=MsoNormal>Audit finding<o:p></o:p></p>

<p class=MsoNormal>Resizing the WYSIWYG text area can only be done by dragging
movements.<o:p></o:p></p>

<p class=MsoNormal>User impact<o:p></o:p></p>

<p class=MsoNormal>Keyboard-only users and many assistive-technology users may
be inconvenienced by not being able to resize the editor<o:p></o:p></p>

<p class=MsoNormal>Remediation plan<o:p></o:p></p>

<p class=MsoNormal>Refactor the interaction so it is fully keyboard operable,
uses correct semantics and states, and preserves predictable focus movement.
Allow the editor to be resized using keyboard.<o:p></o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'>Steps to reproduce</b><o:p></o:p></p>

<p class=MsoNormal>1. Open the form editor containing the WYSIWYG area (For
example, Articles &gt; _article-name_ link &gt; Editor area).<o:p></o:p></p>

<p class=MsoNormal>2. Locate its resize handle.<o:p></o:p></p>

<p class=MsoNormal>3. Attempt to resize it with a pointer.<o:p></o:p></p>

<p class=MsoNormal>4. Attempt the same operation with the keyboard.<o:p></o:p></p>

<p class=MsoNormal>5. Observe that the operation is not keyboard accessible.<o:p></o:p></p>

</body>

</html>

### Version

6.1

### Expected result

Keyboard users can resize editor area.

### Actual result

The resize handle is not keyboard accessible.

### What device were you using? (eg, phone, laptop)

Desktop web

### What web browser were you using? (eg, Chrome, Safari)

Chrome, Firefox, Edge, Safari

### Assistive Technology?

N/A

**Comment by brianteeman** [comment]
The tinymce docs imply this is possible but I cant work out how to actually focus the resize handle and do it even on their own demo

> Focusing the resize handle, and using the arrow keys. The Up Arrow and Left Arrow keys will make the editor smaller, and the Down Arrow and Right Arrow keys will make the editor larger.

**Comment by brianteeman** [comment]
Correction i worked it out

alt+f10 puts the focus on the toolbar. Then you can tab to the resize and use the arrow keys to resize

https://github.com/user-attachments/assets/87d05bbc-69a3-4964-820c-070e75ce4804

**Comment by mbeganyi-a11y** [comment]
Interesting find. I'll look into this. Thanks for the video and nice troubleshooting to find alt+f10.

**Comment by brianteeman** [comment]
I just read the docs :)

**Comment by mbeganyi-a11y** [comment]
So this is good for keyboard accessibility but doesn't quite address the [relevant success criterion (2.5.7)](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html) which indicates that the dragging movement must have a single click equivalent. A suggestion would be something like "Adjust Editor Width" and "Adjust Editor Height" controls to allow a single-click equivalent of the dragging movement contained within the TinyMCE field. Not sure how much jurisdiction you have over the drag handle in the editor, but an alternative to providing these suggested controls would be to remove the drag handle, but of course this means that all users cannot resize their editor field.

**Comment by brianteeman** [comment]
removing the drag handle is just plain stupid. you dont improve accessibility by removing funcrionality just to pass a test nor do you have to if its the only way to achieve it https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html#dfn-essential

you can also maximize the editor at any time with ctrl-shift-f

**Comment by brianteeman** [comment]
Being able to rezise the editor is not something that is essential for the functionality of the editor so the success criterion does not apply

**Comment by mbeganyi-a11y** [comment]
I agree about removing functionality. Was just providing options for posterity's sake. 

I appreciate the keyboard shortcuts but single-pointer equivalents don't include keyboard shortcuts. For example, if an author is editing on a touchscreen device without using a keyboard, they would be required to drag the resize handle. There is no other option.

> Being able to rezise the editor is not something that is essential for the functionality of the editor so the success criterion does not apply

I appreciate the sentiment, but the SC states:: "All [functionality](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html#dfn-functionality) that uses a [dragging movement](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html#dfn-dragging-movement) for operation can be achieved by a [single pointer](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html#dfn-single-pointer) without dragging, unless dragging is [essential](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html#dfn-essential) or the functionality is determined by the [user agent](https://www.w3.org/WAI/WCAG22/Understanding/dragging-movements.html#dfn-user-agent) and not modified by the author." Dragging isn't essential to the editor workflow which thus means it qualifies for this SC. In cases where dragging is essential, a single-pointer equivalent doesn't have to be provided. Since there is functionality for resizing which currently relies on a dragging movement, a single-pointer equivalent must be provided, again, for "all functionality".

**Comment by brianteeman** [comment]
Can you show me an example of any WYSIWYG editor that can be resized in the manner you describe

**Comment by mbeganyi-a11y** [comment]
I can't point to any that ship with those controls but I'm not sure that's relevant. Partial support exists in TinyMCE via the "Fullscreen" option in the "View" menu in the toolbar, but it's not truly sufficient to meet SC 2.5.7 because it only offers one option: make the editor fullscreen. Since there's granularity in the current drag handle implementation, that should be represented by single-pointer equivalents. Now, not ever single pixel dimension needs to be covered, but at least a range of options that fairly represents the granularity is acceptable.

Other tools like CKEditor's "Maximize" plugin also add a fullscreen or maximize option to the toolbar, but again, that only covers one option.

I look at this as a good opportunity for Joomla to provide an accessible solution that it could offer to the community as a resource. Perhaps others have done this work already, but only in proprietary or private systems.

**Comment by brianteeman** [comment]
Knock yourself working on this. I don't agree that it's a requirement as per wcag as this is an optional configuration change and not an essential part of using joomla which all the examples provided by wcag are
